packages/shared/src/authz/policies/canQueryProject.ts

```diff
@@ -18,44 +16,45 @@ import {
   WorkspaceNoAccessError,
   WorkspaceSsoSessionInvalidError
 } from '../domain/authErrors.js'
+import { err, isOk, ok } from 'true-myth/result'
+import { AuthCheckContextLoaderKeys } from '../domain/loaders.js'
+import { LogicError } from '../domain/errors.js'
 
-export const canQueryProjectPolicyFactory =
-  (
-    loaders: Pick<
-      AuthCheckContextLoaders,
-      | 'getEnv'
-      | 'getProject'
-      | 'getProjectRole'
-      | 'getServerRole'
-      | 'getWorkspaceRole'
-      | 'getWorkspaceSsoProvider'
-      | 'getWorkspaceSsoSession'
-    >
-  ) =>
-  async ({
-    userId,
-    projectId
-  }: UserContext & ProjectContext): Promise<
-    AuthResult<
-      | typeof ProjectNotFoundError
-      | typeof ProjectNoAccessError
-      | typeof WorkspaceNoAccessError
-      | typeof WorkspaceSsoSessionInvalidError
-    >
-  > => {
-    const { FF_ADMIN_OVERRIDE_ENABLED, FF_WORKSPACES_MODULE_ENABLED } = loaders.getEnv()
+export const canQueryProjectPolicyFactory: AuthPolicyFactory<
+  | typeof AuthCheckContextLoaderKeys.getEnv
+  | typeof AuthCheckContextLoaderKeys.getProject
+  | typeof AuthCheckContextLoaderKeys.getProjectRole
+  | typeof AuthCheckContextLoaderKeys.getServerRole
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceRole
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceSsoProvider
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceSsoSession,
+  UserContext & ProjectContext,
+  | typeof ProjectNotFoundError
+  | typeof ProjectNoAccessError
+  | typeof WorkspaceNoAccessError
+  | typeof WorkspaceSsoSessionInvalidError
+> =
+  (loaders) =>
+  async ({ userId, projectId }) => {
+    const env = await loaders.getEnv()
+    if (!isOk(env)) {
+      throw new LogicError('Failed to load environment variables')
+    }
+
+    const { FF_ADMIN_OVERRIDE_ENABLED, FF_WORKSPACES_MODULE_ENABLED } = env.value
 
     const project = await loaders.getProject({ projectId })
-    // hiding the project not found, to stop id brute force lookups
-    if (!project) return unauthorized(ProjectNotFoundError)
+    if (!isOk(project)) {
+      return err(project.error)
+    }
 
     // All users may read public projects
     const isPublicResult = await requireExactProjectVisibilityFactory({ loaders })({
       projectId,
       projectVisibility: 'public'
     })
     if (isPublicResult) {
-      return authorized()
+      return ok(true)
```

Comment:
Because we need true-myth structures for loaders (e.g. getProject can not only not find the project, but also just not be able to retrieve it through GQL because the user is not authorized), it also didn't make sense anymore to have our custom authorized()/unauthorized() helpers and I moved to true-myth structs here as well